custom/enikshay/management/commands/investigations_reconciliation.py

```diff
@@ -0,0 +1,258 @@
+
+        properties_by_investigation = defaultdict(dict)
+        values_for_property = defaultdict(list)
+        investigation_case_modified_on = {}
+        coalesced_values = {}
+
+        # iterate over cases to prepare
+        # 1. modified on
+        # 2. values for all case properties
+        # 3. values for all case properties for each investigation case
+        for investigation_case in investigation_cases:
+            investigation_case_modified_on[investigation_case.case_id] = investigation_case.modified_on
+            for case_property in PROPERTIES_TO_BE_COALESCED:
+                prop_value = investigation_case.get_case_property(case_property)
+                # add value for this investigation under its mapping for writing later
+                properties_by_investigation[investigation_case.case_id][case_property] = prop_value
+                # add value for this investigation in the collection for values for investigations
+                values_for_property[case_property].append(prop_value)
+
+        # iterate over all values for each case property to prepare coalesced values
+        for case_property, values in values_for_property.items():
+            coalesced_value = self.get_coalesced_value_for_case_property(
+                case_property, values
+            )
+            coalesced_values[case_property] = coalesced_value
+
+        # write values for each investigation
+        for investigation_case_id, investigation_case_values in properties_by_investigation.items():
+            other_investigation_case_values = investigation_case_values.copy()
+            other_investigation_case_values['investigation_case_id'] = investigation_case_id
+            other_investigation_case_values['modified_on'] = investigation_case_modified_on[investigation_case_id]
+            other_investigation_case_values['episode_id'] = episode_case_id
+            other_investigation_case_values['update_or_close'] = (
+                'to_be_updated'
+                if investigation_case_id == retain_case_id
+                else 'to_be_closed'
+            )
+            self.writerow(other_investigation_case_values)
+
+        # write final coalesced values
+        coalesced_investigation_case = coalesced_values.copy()
+        coalesced_investigation_case['investigation_case_id'] = retain_case_id
+        coalesced_investigation_case['modified_on'] = latest_investigation_case.modified_on
+        coalesced_investigation_case['episode_id'] = episode_case_id
+        coalesced_investigation_case['update_or_close'] = 'update'
+        self.writerow(coalesced_investigation_case)
+
+    def close_or_update_investigation_cases(self, all_cases, retain_case_id, episode_case_id,
+                                            investigation_interval, updates):
+        all_case_ids = [investigation_case.case_id for investigation_case in all_cases]
+        # ToDo: refetch investigation_cases in case the len of set is different from list
```

Comment:
Still actionable?

Reply:
Not worried about this yet. Will look at this when this is needed.